src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -223,6 +223,11 @@ private void UpdateEndpoints()
         {
             var newPathSegments = routePattern.PathSegments.ToList();
             var hasLinkGenerationEndpoint = false;
+            var nonParameterPolicyValues = routePattern.ParameterPolicies
+                .Where(p => routePattern.GetParameter(p.Key ?? string.Empty) == null)
+                .Select(p => new KeyValuePair<string, object>(p.Key, p.Value.First().ParameterPolicy)) // Can only pass a single non-parameter to RouteParameter
```

Comment:
Is this ok? `RoutePatternFactory.Pattern` takes a `RouteValueDictionary` that expects a collection of key/value where the key is the name and the value is the IRouteConstraint. That should be fine because `MapRoute` is the same so there shouldn't be more than one constraint for non-parameter's.

Reply:
Yeah this looks like it's OK.

Reply:
I'd suggest adding a comment that this is only needed because while we've already created a route pattern, we're going to create another one later.